---
layout: post
title: The 6th Argonne Soil Metagenomic Meeting Summary
categories: science, thoughts
tags: meeting
---

The 6th Argonne Soil Metagenomic Meeting/Workshop (#ASMM14) just happened earlier this week (Oct 1-3, 2014) in Pheasant Run Resort, St. Charles, Il. I have been attending this meeting when I was still studying bacteria-metal interactions. Although my main study focus was not soil at the time, I have always found this meeting useful and filled with positive scientific interactions. It was no exception this year. To better conclude on things I have learned, here is my brief summary on soil, science, and the community that studies communities. 

<!--more-->

There were a total of seven sessions span over three days. The program is available [here](http://press3.mcs.anl.gov/asmw14/program/). All of the presentations could be summarized into three broader categories:

1. **Linking ecological functions to microbial communities**   
    This is the largest portion of the meeting. No surprise, it has been every microbial ecologist's dream to understand the impacts of microbial community and its residing environment and it has been every modelist's dream to predict these impacts emprically. Well, we are not there yet but we are making a good progress! We frequently trying to corealte environmental factors, such as pH, soil moisture, soil density, etc. to the distribution of microbial communities. The importance of slope and relationship to sun exposure have also been introduced (Dr. James Tiedje and [Aurora MacRae-Crerar](https://twitter.com/auroralexandra)). It was also interesting to see how little we know. [Ryan Williams](http://ryanw.public.iastate.edu/Ryan_J_Williams_Homepage/Home_Page.html) showed that the soil enzymes biogeochemists measure commonly in lab were infact not as common as we think in the soil. [Aaron Garoutte](https://twitter.com/AaronGaroutte) showed that approximately 60% of assembled soil microbial metatranscripts were annotated as hypothetical protein encoding genes. I know, it's a bit of disencouraging. But many have shown that we are not doomed and it is possible to link microbial communities and ecological functions, especially when combining ecological parameters with different "omics" tools. Out of all of them, my favorites (with some current project bias) are [Dr. Kristen DeAngelis'](https://twitter.com/kristenobacter) talk on how microbial communities responded to Harvard Forest climate change experiment over two decades, Dr. Laurent Philippot's presentation on linking microbial diversity and nitrogen cycle, Dr. Janet Jansson's multi "omic" approach in trying to decode microbial life styles in bog and permafrost, and Dr. Pet Baldrian's take on microbial (especially fungal) activities on degradation in a temporate forest. It was absolutely encouraging to see [Grace Pold's](https://twitter.com/GPFriedSci) metatranscriptomic data showing that extra cellular protease encoding genes increased as soil temperated increased in Harvard forest. I also got really excited when Mary-Cathrine Leewis showed the high abundance of chitin and lignin genes in boreal forests.  
 
2. **Tool development**  
    There were several tool developments caught my eyes at the meeting. I really liked how Dr. Ondřej Uhlík incorporated [RDP FrameBot](https://rdp.cme.msu.edu/tutorials/framebot/RDPtutorial_FRAMEBOT.html) to mine hydroxylating dioxygenase genes de novo. Dr. Patric Chain's definition of signature gene fragments in metagenome and how he uses these fragments for community comparisons. For a simple diversity measure, [Qingpeng Zhang's](http://www.cse.msu.edu/~qingpeng/) [informational genomic segment](http://nbviewer.ipython.org/github/qingpeng/2013-diversity/blob/master/notebook/2013-diversity-summary.ipynb) was brilliant for complex community. [Will Trimble](https://github.com/wltrimbl) also introduced a kmer method to eliminate errors in metagenome. During the discussion, Will's kmer method was frequently brought up to compare to [Nonpareil](http://bioinformatics.oxfordjournals.org/content/30/5/629.long). My limited understanding of the difference is Nonpareil relies on curve fitting, where Will's kmer method depends on drastic kmer abundance diferences.   
    Binning was another big part of the meeting. [MetaBAT](https://bitbucket.org/berkeleylab/metabat) and [MaxBin](http://www.microbiomejournal.com/content/2/1/26) were introduced for multiple (MetaBAT) and single (MaxBin) metagenome partitioning. Both methods use read abundance and tetranucleotide frequency to predict sequences there are likely originated from the same genome. Similarly, Dr. Titus Brown's [partitioning method](http://www.pnas.org/content/111/13/4904) appeared to be effective in separating reads into species specific bins. I would love to see a benchmark comparison on how these three tools perform on a collection of mock metagenomic data sets.    
    In addition to bioinfomatic tools, a couple of databases have to be mentioned. One is [FOAM: Functional Ontology Assignments for Metagenomes](http://portal.nersc.gov/project/m1317/FOAM/index.php) and the other one is RefSoil (a mannually curated soil-specific reference genome database). I don't ahve the link RefSoil yet but I can see myself using both databases extensively in the near future.  
    One last tool I have to mention is Dr. Stefan Green's two stage PCR method on tackeling primer introduced bias. The paper with data is submitted and being reviewed. I'm really intrigued by the method and would love to see the paper when it comes out.   

3. **Back to model microbes**   
    From a person who used to work with bacterial isolates, it was truely exciting to see how isolates are working their way back into the community. Genome comparason is an effective way of evaluate the evolution and niche history of an organisms. Melissa Dsouza presented how Antarctic *Paenibacillus* genome was reduced when comparing to *Paenibacillus* found in temperate environments. I personally believe there is a great potential in isolating novel microbes based on metagenome results for model community assemblage studies.   

To summarize, the meeting was a success (I know, it's a "duh..." comment). The theme of the meeting was definitely "Actinobacteria", "multi-omics", and "community efforts." As a microbial ecologist who moved from sediment and water environment to soil, I frequently conclude that soil is one of the most complex thing I will ever deal with in my life. I believe many feels my pain one way or the other. Hence, it is important to tackel the complexity from something small and one thing at a time. [Dr. Adina Howe](germs-lab.github.io) and [Dr. Erick Cardenas](https://twitter.com/erickcardenasp) proposed to compose a soil only genome/metagenome database curated by community (extending from RefSoil), which hopefully would grow into a project like [Human Microbiome Project](http://www.hmpdacc.org/) and benefit the entire soil community. While there may be many challenges down the road, it is absolutely wonderful to see how enthusiastic the participants were. To see scientists with multidisciplinary background coming together for solve one big science puzzle is definitely what attracted me to this meeting from year to year. I can't wait to see everyone next year! :)
